src/Facility.Definition/Fsd/FsdParser.cs

```diff
@@ -86,21 +110,14 @@ public ServiceInfo ParseDefinition(NamedText source)
 					.ThenByDescending(x => x.LineColumn.ColumnNumber)
 					.First();
 
-				throw new ServiceDefinitionException(
+				service = null;
+				errors.Add(new ServiceDefinitionError(
 					"expected " + string.Join(" or ", expectation.Names.Distinct().OrderBy(GetExpectationNameRank).ThenBy(x => x, StringComparer.Ordinal)),
 					new NamedTextPosition(source.Name, expectation.LineColumn.LineNumber, expectation.LineColumn.ColumnNumber),
-					exception);
-			}
-
-			// check for unused remarks sections
-			foreach (var remarksSection in remarksSections.Values)
-			{
-				string sectionName = remarksSection.Name;
-				if (service.Name != sectionName && service.FindMember(sectionName) == null)
-					throw new ServiceDefinitionException($"Unused remarks heading: {sectionName}", remarksSection.Position);
+					exception));
 			}
 
-			return service;
+			return errors.Count == 0;
```

Comment:
Should we `null` the `service` if there are any errors?

Reply:
The idea is that the service can be returned, even if semantically invalid, so that it can be used for completion by the language service.